harness-delegate-ng/templates/deployment.yaml

```diff
@@ -33,7 +33,7 @@ spec:
         fsGroup: 1001
       containers:
         - name: delegate
-          image: {{ .Values.delegateDockerImage }}
+          image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
```

Comment:
why not use existing variable? what you did makes sense but this will break other customers. 
If we have have to go this way, I'll say write a function such that.
if (image exists)
 return {{ .Values.image.repository }}:{{ .Values.image.tag }}
else 
 return {{ .Values.delegateDockerImage }}

Reply:
just bump the chart version to 2.0.0 to announce the backward compatibility. 

this is the standard way of setting image repo and tag across all of the CNCF charts.

Reply:
@djsly a non backward compatible change here, will have high blast radius for us and it will create issues for a lot of customers. Can we use existing set of variables instead?

Reply:
@mdmilic it's a bit concerning to read that the monthly images are not making it to the Chart. Having your chart published to artifacthub makes it such that users are using this central view on the status of the chart.  Normally, the expectation is that when a new appVersion is released, a matching chart version sees the day using that new image version. We cannot use the auto upgrader since all changes needs to follow a safe deployment practice and the upgrade of the delegate image need to be staged properly and tied to a specific release on our side.  What we do is we monitor artifacthub for new available charts and we scan the security results of those, we decide when to pick up the new charts based on many factor from this point on, but we tend to upgrade charts at least once a month with this mechanisms.

Reply:
Hi @djsly, 

This is related to the way we release delegates. We don't release delegate to all users at once, we release it in staged manner over a period of time. On the other hand chart, when released, is available to all customers at the same time. This is why we don't release chart with every latest delegate release (technically we could release it for every version once that version propagates to all customers, but that would still put the chart behind, trigger automatic upgrades immediately and we didn't find it much helpful). 
Customers are always welcome to pick and choose their delegate version (and might choose the newer version than what we have released for them), but for majority that just leave the process automatic, we handle the upgrades according to the above mentioned staged process. 
If you want to have semi-automatic upgrades, we also provide an API endpoint which you can use to query what's the latest default delegate version for your account and you can use that to feed into the chart (e.g. create a script or a pipeline that does the upgrade). 
Hope this alleviates any concerns you had regarding the release process (please let me know if you have any other). Will leave chart review to @AnupamIO 

Thanks,
Marko